bin/install

```diff
@@ -57,7 +60,11 @@ function install_rabbitmq {
 	# If rabbitmq $version >= 3.6.0
 	if [[ ${_semver//.} -ge '360' ]]; then
 		url=https://github.com/rabbitmq/rabbitmq-server/releases/download/$version/$os_prefix-$_semver.tar.xz
-		curl -L $url --output $tmp_download_dir/rabbitmq.tar.xz
+		status=$(curl --write-out %{http_code} -L $url --output $tmp_download_dir/rabbitmq.tar.xz)
```

Comment:
```suggestion
		status=$(curl  --write-out %{http_code} -fL $url --output $tmp_download_dir/rabbitmq.tar.xz)
```

This way you don't need to do the 200 check below.

I think you may need to have `set -o pipefail` set but not 100% for this to work—though we should add that anyways.

Reply:
Neat, but would't it be better to have a more specific error than `The requested URL returned error: 404`?

Reply:
I feel like I'd like to see whatever the server said even if it's not the most user-helpful message. That way if it's something else like a rate limit or something the user gets a more accurate message.

If you want to go the extra mile, perhaps there is a way to specifically handle 404's with a custom message but personally I don't think it's worth the effort.